extensions/commands/cci/cmd_export_all_versions.py

```diff
@@ -0,0 +1,131 @@
+import yaml
+
+from conan.api.output import ConanOutput
+from conan.cli.command import conan_command, OnceArgument
+from conan.errors import ConanException
+
+
+def output_json(result):
+    print(json.dumps({
+        "exported": [repr(r) for r in result['exported']],
+        "failures": result['failures']
+    }))
+
+
+def output_markdown(result):
+    exported = result['exported']
+    failures = result['failures']
+    print(textwrap.dedent(f"""
+    ### Conan Export Results
+
+    Successfully exported {sum([len(versions) for versions in exported.values()])} versions from {len(exported.keys())} recipes while encountering {len(failures)} recipes that could not be exported; these are
+
+
+    <table>
+    <th>
+    <td> Recipe </td> <td> Reason </td>
+    </th>"""))
+
+    for key, value in failures.items():
+        print(textwrap.dedent(f"""
+            <tr>
+            <td> {key} </td>
+            <td>
+
+            ```txt
+            """))
+        print(f"{value}")
+        print(textwrap.dedent(f"""
+            ```
+
+            </td>
+            </tr>
+            """))
+
+    print("</table>")
+
+
+@conan_command(group="Conan Center Index", formatters={"text": lambda result: None, "json": output_json, "md": output_markdown})
+def export_all_versions(conan_api, parser, *args):
+    """
+    Export all version of either a single recipe, a list of recipes, or a global recipes folder
+    """
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('-n', '--name', action=OnceArgument, help="Name of the recipe to export")
+    group.add_argument('-l', '--list', action=OnceArgument, help="YAML file with list of recipes to export")
+    group.add_argument('-p', '--path', action=OnceArgument, help="Path of the global recipe folder")
+    args = parser.parse_args(*args)
+
+    if args.name is not None:
+        recipes_to_export = [args.name]
+    elif args.list is not None:
+        recipes_to_export = yaml.safe_load(open(args.list, 'r'))["recipes"]
+    else:
+        recipes_to_export = os.listdir(args.path)
+
+    out = ConanOutput()
+
+    # Result output variables, these should always be returned
+    exported = {}
+    exported_refs = []
+    exported_with_revision = []
+    failed = dict()
+
+    for item in recipes_to_export:
+        recipe_name = item if not isinstance(item, dict) else list(item.keys())[0]
+        folders_to_export = item[recipe_name][0]['folders'] if isinstance(item, dict) else None
+        out.verbose(f"Processing recipe '{recipe_name}'")
+
+        recipe_folder = os.path.join("recipes", recipe_name)
+        if not os.path.isdir(recipe_folder):
+            raise ConanException(f"Invalid user input: '{recipe_name}' folder does not exist")
+
+        config_file = os.path.join(recipe_folder, "config.yml")
+        if not os.path.isfile(config_file):
+            raise ConanException(f"The file {config_file} does not exist")
+
+        with open(config_file, "r") as file:
```

Comment:
I'd try to avoid extra nesting with:
```python
config = yaml.safe_load(open(config_file, "r").read())
```
or the like. But having all the processing of the contents inside the ``open`` context seems a bit too much.